test/lib/absinthe/extensions_test.exs

```diff
@@ -0,0 +1,47 @@
+defmodule Absinthe.ExtensionsTest do
+  use Absinthe.Case, async: true
+
+  defmodule Schema do
+    use Absinthe.Schema
+
+    query do
+      field :foo, :string do
+        middleware :resolve_foo
+      end
+    end
+
+    def resolve_foo(res, _opts) do
+      %{res |
+        value: "hello world",
+        state: :resolved,
+        extensions: %{foo: 1},
+      }
+    end
+  end
+
+  defmodule MyPhase do
+    # rolls up the extensions data into a top level result
+    def run(blueprint, _) do
+      extensions = get_ext(blueprint.resolution.result.fields)
+      result = Map.put(blueprint.result, :extensions, extensions)
```

Comment:
The extension data is not included in the result by default because it is not possible to know how to roll that data up. Thus a phase must be provided to do so.